### PR TITLE
 Add arm64,armv7 - Add version tag - update docker and python req

### PR DESCRIPTION
**Pull request by 1337-server**
- Adds check to only push docker image when it pushed (not on pr like it currently does)
- Adds Dependabot (Checks docker, pip and python requirements)
- Adds Ubuntu 18 compatibility checks
- Adds the arm architectures (arm64, armv7, armhf) to the build, allowing A.R.M to be run on arm.
   - Added QEMU 
   - Added step to show platform versions available
   - Automatically reads the version from the new VERSION file and pushes to the version & latest tags
   - This significantly increases the build times (Average is 1h)
- Added  `libzbar-dev libzbar0 scons swig` for arm64 dependencies
- Build now compiles MakeMKV ensuring the latest version

- Updated all links to now use the main arm repo
- Bumped version (1.0.3)

### Version bumps
- [Bump beautifulsoup4 from 4.10.0 to 4.11.1](https://github.com/1337-server/arm-dependencies/commit/901f64457d77673a2d96cad7adc91cbab5ee327c)
- [Bump soupsieve from 2.3.1 to 2.3.2.post1 (](https://github.com/1337-server/arm-dependencies/commit/df63d3e206fda5b417f3287c7f56cae628a7d318)
- [Bump cffi from 1.15.0 to 1.15.1 (](https://github.com/1337-server/arm-dependencies/commit/f3aca97bc978d2e3357f32b41783044bb4091835)
- [Bump flask-wtf from 1.0.0 to 1.0.1 (](https://github.com/1337-server/arm-dependencies/commit/24283a8979380f0011ef0b810641c6fe7f7b50b7)
- [Bump xmltodict from 0.12.0 to 0.13.0 (](https://github.com/1337-server/arm-dependencies/commit/cdc92461cf1c527ab210f683a52e3a8f583a0280)
- [Bump sqlalchemy from 1.4.32 to 1.4.39 (](https://github.com/1337-server/arm-dependencies/commit/e9c08f6f790d30b2d396c303396784b7dd65f874)
- [Bump psutil from 5.9.0 to 5.9.1 (](https://github.com/1337-server/arm-dependencies/commit/4c830553911d050f72776a699aa1d5c879672ab0)
- [Bump apprise from 0.9.7 to 0.9.9 (](https://github.com/1337-server/arm-dependencies/commit/3d1e4e5fe426fdc15fededade8d32c6bfdb8ea34)
- [Bump urllib3 from 1.26.9 to 1.26.10 (](https://github.com/1337-server/arm-dependencies/commit/00ab6de0f7c51fed12303708bee7b0f30988fcb1)
- [Bump certifi from 2021.10.8 to 2022.6.15 (](https://github.com/1337-server/arm-dependencies/commit/80c4cc36357a1a8e54020c7214a891c35f3ee079)
- [Bump actions/setup-python from 2 to 4 (](https://github.com/1337-server/arm-dependencies/commit/b478e5ef251e6ed0ac86893037090b526e4e9369)

### CODE REVIEW

File: .github/workflows/publish-image.yml [hunk 1]
```diff
@@ -72,26 +82,42 @@ jobs:
...
+                uses: docker/build-push-action@v3
                 with:
                     context: .
-                    push: true
-                    tags: ${{ env.IMAGE_NAME }}:${{ env.TAG }}
+                    platforms: linux/amd64,linux/arm64,linux/arm/v7
+                    push: ${{ github.event_name != 'pull_request' }}
+                    tags: |
+                      ${{ env.IMAGE_NAME }}:${{ env.TAG }}
+                      ${{ env.IMAGE_NAME }}:${{ env.VERSION }}
```
Comment (1337-server) [inline .github/workflows/publish-image.yml:110]: Push to both the branch tag (using `latest` for the default branch) and the version tag from the contents of the version file

non default branches version file should be updated with non-standard semantic versions, e.g. 
- `dev-1.0.1`
- `indev-1.0.1`
- etc

As using normal versions will overwrite the default versions, for example.
A version 1.0.1 inside the indev branch will overwrite the main branch 1.0.1 on docker hub if allowed to push.

File: .github/workflows/publish-image.yml [hunk 2]
```diff
@@ -72,26 +82,42 @@ jobs:
...
                     labels: ${{ steps.meta.outputs.labels }}
+                    build-args:
+                      BUILD_DATE=${{ env.BUILD_DATE }}
+                      VERSION=${{ env.VERSION }}
 
             -
                 name: retag and push multiplatform images to multiple registries
+                if: github.event_name != 'pull_request'
```
Comment (1337-server) [inline .github/workflows/publish-image.yml:118]: Only push if its not a PR

File: .github/workflows/publish-image.yml [hunk 3]
```diff
@@ -72,26 +82,42 @@ jobs:
...
                 uses: akhilerm/tag-push-action@v2.0.0
                 with:
                     src: docker.io/${{ env.IMAGE_NAME }}:${{ env.TAG }}
                     # make sure the github name is correct
-                    dst: ${{ env.REGISTRY }}/${{ github.actor }}/arm-dependencies:${{ env.TAG }}
+                    dst: ${{ env.REGISTRY }}/${{ github.repository_owner }}/arm-dependencies:${{ env.TAG }},${{ env.REGISTRY }}/${{ github.repository_owner }}/arm-dependencies:${{ env.VERSION }}
```
Comment (1337-server) [inline .github/workflows/publish-image.yml:123]: Push to both branch tag and version

also update github.actor

File: .github/workflows/ubuntu18.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        python-version: ${{ matrix.python-version }}
+    - name: Update and Upgrade
+      run: |
+        sudo apt update && sudo apt upgrade -qqy
+        sudo apt install libcurl4-openssl-dev libssl-dev
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8 pytest
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
```
Comment (1337-server) [inline .github/workflows/ubuntu18.yml:37]: Run setup for Ubuntu 18, run a matrix checking 4 python versions for pip dependencies and fail check if any deps can't be found/installed.

This help quickly check if a new updated package can't be found on specific python version on Ubuntu 18.

File: .github/workflows/version_bump.yml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+          git config --global user.email "dependabot[bot]@users.noreply.github.com"
+          git commit -m "[Automated] Increment Version"
+        working-directory: actions-runner
+
+      - name: Push increment version commit
+        if: ${{ !steps.check_modified.outputs.version_modified && github.event_name == 'pull_request' && github.actor == 'dependabot[bot]'}}
+        uses: ad-m/github-push-action@v0.6.0
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          branch: ${{ github.head_ref }}
```
Comment (1337-server) [inline .github/workflows/version_bump.yml:59]: This is taken from the Dependabot GitHub.

It's designed to automatically bump the version when Dependabot makes a PR, it doesn't work currently and will need more time of testing/debugging.

File: Dockerfile [hunk 6]
```diff
@@ -79,21 +79,31 @@ RUN \
...
+COPY ./scripts/install_makemkv.sh /install_makemkv.sh
+RUN chmod +x /install_makemkv.sh && sleep 1 && \
+    /install_makemkv.sh
 # clean up apt
 RUN apt clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
+# Container healthcheck
+COPY scripts/healthcheck.sh /healthcheck.sh
+RUN chmod +x /healthcheck.sh
+HEALTHCHECK --interval=5m --timeout=15s --start-period=30s CMD /healthcheck.sh
```
Comment (1337-server) [inline Dockerfile:98]: Copy healthcheck and run it every 5 mins

Allows arm to show unhealthy if anything is broken inside the container

**Comment by shitwolfymakes** [comment]
Fantastic work!

**Review by shitwolfymakes — CHANGES_REQUESTED** [review]

File: .github/dependabot.yml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+      interval: "daily"
+    # Allow up to 10 open pull requests for pip dependencies
+    open-pull-requests-limit: 20
+  # Maintain dependencies for pip
+  - package-ecosystem: "pip"
+    directory: "/"
+    schedule:
+      interval: "daily"
+    # Allow up to 10 open pull requests for pip dependencies
+    open-pull-requests-limit: 20
```
[the remark this answers is not in the document]
Comment (1337-server) [reply .github/dependabot.yml:28]: If it's updated frequently, 10 should be fine. But for the initial PR, 10 wasn't enough.

[the remark this answers is not in the document]
Comment (shitwolfymakes) [reply .github/dependabot.yml:28]: Sounds good

**Review by shitwolfymakes — APPROVED** [review]

**Pull request merged**